actions/microsoft-onedrive/microsoft_onedrive/onedrive_mutate_action.py

```diff
@@ -0,0 +1,258 @@
+import os
+from pathlib import Path
+from typing import List, Literal
+
+import requests
+from sema4ai.actions import ActionError, OAuth2Secret, Response, action
+
+from microsoft_onedrive.models import (
+    DeleteOneDriveItemParams,
+    OneDriveFolderCreationParams,
+    OneDriveUploadRequest,
+    RenameOneDriveItemParams,
+)
+from microsoft_onedrive.support import (
+    BASE_GRAPH_URL,
+    build_headers,
+)
+
+
+@action
+def create_onedrive_folder(
+    token: OAuth2Secret[
+        Literal["microsoft"],
+        List[Literal["Files.ReadWrite"]],
+    ],
+    params: OneDriveFolderCreationParams,
+) -> Response[dict]:
+    """
+    Create a new folder in OneDrive.
+
+    Args:
+        token: OAuth2 token to use for the operation.
+        params: OneDriveFolderCreationParams object containing:
+            folder_name: name of the folder to create.
+            parent_folder_path: path of the parent folder in OneDrive (optional).
+
+    Returns:
+        Response with the result of the operation
+    """
+    headers = build_headers(token)
+
+    if params.parent_folder_path:
+        create_url = (
+            f"{BASE_GRAPH_URL}/me/drive/root:/{params.parent_folder_path}:/children"
+        )
+    else:
+        create_url = f"{BASE_GRAPH_URL}/me/drive/root/children"
+
+    folder_data = {
+        "name": params.folder_name,
+        "folder": {},
+        "@microsoft.graph.conflictBehavior": "rename",
+    }
+
+    create_response = requests.post(create_url, headers=headers, json=folder_data)
+
+    if create_response.status_code in [200, 201]:
```

Comment:
I think a `create_response.raise_for_status()` would simplify things and remove the ifs.